features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -169,25 +169,31 @@ internal class Base64Serializer private constructor(
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ): Bitmap? {
-        var result: Bitmap? = null
         if (shouldUseDrawableBitmap(drawable)) {
-            drawableUtils.createScaledBitmap(
-                (drawable as BitmapDrawable).bitmap
-            )?.let { scaledBitmap ->
-                val shouldCacheBitmap = scaledBitmap != drawable.bitmap
-
-                serializeBitmapAsynchronously(
-                    drawable,
-                    scaledBitmap,
-                    shouldCacheBitmap,
-                    imageWireframe,
-                    base64SerializerCallback
-                )
-
-                result = scaledBitmap
-            }
+            val bitmap = (drawable as BitmapDrawable).bitmap
+
+            Runnable {
+                @Suppress("ThreadSafety") // this runs inside an executor
+                drawableUtils.createScaledBitmap(
+                    bitmap
+                )?.let { scaledBitmap ->
+                    val shouldCacheBitmap = scaledBitmap != drawable.bitmap
```

Comment:
I have the same question here...it seems that `createScaledBitmap` always creates a new instance so how can this condition be false ?

Reply:
If createScaledBitmap is called with the same dimensions as the existing bitmap then it returns the instance of the existing bitmap. I can also compare width/height of the bitmaps if it would be more performant than comparing by reference